AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCoreDataClient.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+#import "MSBaseOptions.h"
+#import "MSDataStore.h"
+#import "MSDocumentStore.h"
+#import "MSTokensResponse.h"
+#import "MS_Reachability.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^MSCachedTokenCompletionHandler)(MSTokensResponse *_Nullable tokens, NSError *_Nullable error);
+
+/**
+ * Class responsible for the core logic between offline and remote calls.
+ */
+@interface MSCoreDataClient : NSObject
```

Comment:
Can we rename this class? It is a bit confusing with this: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CoreData/index.html

Reply:
Agreed, we changed it to `MSDataOperationProxy`.